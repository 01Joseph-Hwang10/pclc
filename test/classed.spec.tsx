/* eslint-disable react/react-in-jsx-scope */
import clsx from "clsx";
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { FunctionalComponent, h } from "preact";
import classed, { tw } from "../dist";
import { render } from "@testing-library/preact";

interface ImageProps {
  src: string;
  alt?: string;
  className?: string;
}

const Image: FunctionalComponent<ImageProps> = ({ src, alt, className }) => {
  return (
    <img
      src={src}
      alt={alt}
      className={clsx([
        "flex justify-center align-center rounded shadow-md",
        className,
      ])}
    />
  );
};

const Wrapper = classed.div("h-full flex flex-col");
const Title = classed.div`text-gray-500 text-center text-xs py-2 border-b`;
const CarList = classed.div`flex flex-col flex-1 overflow-scroll`;
const CarItem = classed.div<{ selected?: boolean }>(({ selected }) =>
  clsx([
    tw`border-2 flex p-3 m-2 items-center`,
    selected ? tw`border-black` : tw`border-white`,
  ])
);
const CarImage = classed<ImageProps>(Image)`h-14`;
const CarCaption = classed.span`text-gray-500 text-xs py-2 text-center`;

const App: FunctionalComponent = () => (
  <Wrapper>
    <Title>Choose a ride, or swipe up for more</Title>
    <CarList>
      <CarItem selected={true}>
        <CarImage src="https://via.placeholder.com/100" />
        <CarCaption>Car 1</CarCaption>
      </CarItem>
      <CarItem>
        <CarImage src="https://via.placeholder.com/100" />
        <CarCaption>Car 2</CarCaption>
      </CarItem>
    </CarList>
  </Wrapper>
);

describe("classed", () => {
  it("should match with snapshot when used classed-components", () => {
    const { container } = render(<App />);
    expect(container.innerHTML).toMatchSnapshot();
  });
});
