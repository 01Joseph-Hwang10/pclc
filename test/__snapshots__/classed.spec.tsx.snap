// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classed should match with snapshot when used classed-components 1`] = `"<div class=\\"h-full flex flex-col\\"><div class=\\"text-gray-500 text-center text-xs py-2 border-b\\">Choose a ride, or swipe up for more</div><div class=\\"flex flex-col flex-1 overflow-scroll\\"><div class=\\"border-2 flex p-3 m-2 items-center border-black\\" selected=\\"true\\"><img src=\\"https://via.placeholder.com/100\\" class=\\"flex justify-center align-center rounded shadow-md h-14\\"><span class=\\"text-gray-500 text-xs py-2 text-center\\">Car 1</span></div><div class=\\"border-2 flex p-3 m-2 items-center border-white\\"><img src=\\"https://via.placeholder.com/100\\" class=\\"flex justify-center align-center rounded shadow-md h-14\\"><span class=\\"text-gray-500 text-xs py-2 text-center\\">Car 2</span></div></div></div>"`;
